pybgpstream/pybgpstream.py

```
@@ -101,6 +101,11 @@ def _maybe_add_filter(self, fname, f_single, f_list):
     def _datestr_to_epoch(datestr):
         if datestr is None:
             return 0
+        if isinstance(datestr, int):
+            return datestr
+        assert(isinstance(datestr, str))
```

I think this should be try/except?

---

this function takes either a string or a integer. Any other type should not be acceptable. I'd like it to crash if the user feed some strange type to it. try/except seems too generous here.

---

sorry, i wasn't thinking, i meant you want to raise an exception if this is the case.
i think the correct exception would be a TypeError (https://docs.python.org/3/library/exceptions.html#TypeError)

---

or, probably it would be better to just remove the string check completely and let the date parser deal with whatever we give it